### ISSUE TITLE
Customizing error handlers

### ISSUE THREAD

**Issue by bbarenblat** [urweb/urweb#39]
`urweb` should allow users to customize all error handlers, not just the 500 error, so that error pages can be styled appropriately.

**Comment by achlipala** [comment]
Sure, sounds helpful enough, though not a high enough priority that I'm likely to tackle it soon.

**Comment by wisn** [comment]
@bbarenblat Is this still going?

**Comment by achlipala** [comment]
My guess is no one is working on it, making it a great entry point for a new contributor to the Ur/Web code base!

**Comment by wisn** [comment]
@achlipala I will give it a try. May you help me as where to start?

https://github.com/urweb/urweb/blob/6273c4602a8103f23856616966c34721ad726d3e/src/c/http.c#L57-L59

I'm trying to search any file that contains `500` and I found this one. My guess is right there.

**Comment by achlipala** [comment]
I don't think that function contains a use of the app-specific error handler, and also, as the filename reveals, you are looking at HTTP-specific code, while Ur/Web supports other protocols, too, via a plugin system.  Instead, search `src/c/urweb.c` for `on_error`, to see use of a custom error handler tucked away in an application's configuration.

Hooking 404 errors, in particular, requires a bit more involved process.  You can see in `src/c/cjr_print.sml` the code that *generates* the code that signals a 404 error.  (Just search for "404"!)  Presumably the generator code should instead generate code that calls the app-specific 404 handler, which in turn is probably configured by a new or expanded `.urp`-file directive, handling of which is most concentrated in `src/compiler.sml`.
